test/e2e/recovery_and_restart/recovery_and_restart_test.go

```diff
@@ -0,0 +1,161 @@
+)
+
+var _ = ginkgo.Describe("", func() {
+	var kClient k8s.KubeCtl
+	var restClient yunikorn.RClient
+	var oldConfigMap *v1.ConfigMap
+	var sleepRespPod *v1.Pod
+	var dev = "dev" + common.RandSeq(5)
+
+	// Define sleepPod
+	sleepPodConfigs := common.SleepPodConfig{Name: "sleepjob", NS: dev}
+	sleepPod2Configs := common.SleepPodConfig{Name: "sleepjob2", NS: dev}
+
+	ginkgo.BeforeSuite(func() {
+		// Initializing kubectl client
+		kClient = k8s.KubeCtl{}
+		Ω(kClient.SetClient()).To(gomega.BeNil())
+		// Initializing rest client
+		restClient = yunikorn.RClient{}
+
+		ginkgo.By("Enable basic scheduling config over config maps")
+		var c, err = kClient.GetConfigMaps(configmanager.YuniKornTestConfig.YkNamespace,
+			configmanager.DefaultYuniKornConfigMap)
+		Ω(err).NotTo(gomega.HaveOccurred())
+		Ω(c).NotTo(gomega.BeNil())
+
+		oldConfigMap = c.DeepCopy()
+		Ω(c).Should(gomega.BeEquivalentTo(oldConfigMap))
+
+		// Define basic configMap
+		configStr, err2 := common.CreateBasicConfigMap().ToYAML()
+		Ω(err2).NotTo(gomega.HaveOccurred())
+
+		c.Data[configmanager.DefaultPolicyGroup] = configStr
+		var d, err3 = kClient.UpdateConfigMap(c, configmanager.YuniKornTestConfig.YkNamespace)
+		Ω(err3).NotTo(gomega.HaveOccurred())
+		Ω(d).NotTo(gomega.BeNil())
+
+		ginkgo.By("create development namespace")
+		ns1, err := kClient.CreateNamespace(dev, nil)
+		gomega.Ω(err).NotTo(gomega.HaveOccurred())
+		gomega.Ω(ns1.Status.Phase).To(gomega.Equal(v1.NamespaceActive))
+
+		ginkgo.By("Deploy the sleep pod to the development namespace")
+		sleepRespPod, err = kClient.CreatePod(common.InitSleepPod(sleepPodConfigs), dev)
+		gomega.Ω(err).NotTo(gomega.HaveOccurred())
+		// Wait for pod to move to running state
+		err = kClient.WaitForPodBySelectorRunning(dev,
+			fmt.Sprintf("app=%s", sleepRespPod.ObjectMeta.Labels["app"]),
+			10)
+		gomega.Ω(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Restart the scheduler pod")
+		schedulerPodName, err := kClient.GetSchedulerPod()
+		Ω(err).NotTo(gomega.HaveOccurred())
+		err = kClient.DeletePod(schedulerPodName, configmanager.YuniKornTestConfig.YkNamespace)
+		Ω(err).NotTo(gomega.HaveOccurred())
+		err = kClient.WaitForPodBySelectorRunning(configmanager.YuniKornTestConfig.YkNamespace, fmt.Sprintf("component=%s", configmanager.YKScheduler), 10)
+		Ω(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Port-forward scheduler pod after restart")
+		// kill running kubectl port-forward process if it exists
+		kClient.KillPortForwardProcess()
+		// port-forward the scheduler pod
+		err = kClient.PortForwardYkSchedulerPod()
+		Ω(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Deploy 2nd sleep pod to the development namespace")
+		sleepRespPod2, err := kClient.CreatePod(common.InitSleepPod(sleepPod2Configs), dev)
+		gomega.Ω(err).NotTo(gomega.HaveOccurred())
+		// Wait for pod to move to running state
+		err = kClient.WaitForPodBySelectorRunning(dev,
+			fmt.Sprintf("app=%s", sleepRespPod2.ObjectMeta.Labels["app"]),
+			10)
```

Comment:
this as well, probably 60 seconds, since it may take some time for recovery to complete.